app/controllers/signup.js

```diff
@@ -166,6 +176,38 @@ export default class SignupController extends Controller {
     }
   }
 
+  async checkUserName(userName) {
+    if (!userName) {
+      return set(this.fields[2], 'errorMessage', 'Username cannot be empty');
+    }
+    try {
+      const response = await fetch(
+        `${BASE_URL}/users/isUsernameAvailable/${userName}`,
+        {
+          method: 'GET',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          credentials: 'include',
+        }
+      );
+      const data = await response.json();
+      const { isUsernameAvailable } = data;
+      set(this.fields[2], 'showError', !isUsernameAvailable);
+      set(
+        this.fields[2],
+        'errorMessage',
+        `${userName} is not available, Choose another username!`
+      );
```

Comment:
We must be converting the username to lowercase first and then perform the checks